tests/benchmarks/cpu/find_volume.cpp

```diff
@@ -0,0 +1,80 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021-2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s).
+#include "detray/detectors/create_toy_geometry.hpp"
+
+// VecMem include(s).
+#include <vecmem/memory/host_memory_resource.hpp>
+
+// Google include(s).
+#include <benchmark/benchmark.h>
+
+// System include(s).
+#include <iostream>
+
+// Benchmarks the cost of searching a volume by position
+void BM_FIND_VOLUMES(benchmark::State &state) {
+
+    // This test is broken at the moment, don't run it.
+    state.SkipWithError("Benchmark disabled for the toy geometry");
+    return;
+
+    // Detector configuration
+    static constexpr unsigned int n_brl_layers{4u};
+    static constexpr unsigned int n_edc_layers{7u};
+    vecmem::host_memory_resource host_mr;
+    auto d = detray::create_toy_geometry(host_mr, n_brl_layers, n_edc_layers);
+
+    static const unsigned int itest = 10000u;
+
+    auto &volume_grid = d.volume_search_grid();
+
+    const auto &axis_r = volume_grid.get_axis<detray::n_axis::label::e_r>();
+    const auto &axis_z = volume_grid.get_axis<detray::n_axis::label::e_z>();
+
+    // Get a rough step size from irregular axes
+    auto range0 = axis_r.span();
+    auto range1 = axis_z.span();
+
+    detray::scalar step0{(range0[1] - range0[0]) / itest};
+    detray::scalar step1{(range1[1] - range1[0]) / itest};
+
+    std::size_t successful{0u};
+    std::size_t unsuccessful{0u};
+
+    for (auto _ : state) {
+        for (unsigned int i1 = 0u; i1 < itest; ++i1) {
+            for (unsigned int i0 = 0u; i0 < itest; ++i0) {
+                __plugin::vector3<detray::scalar> rz{
+                    static_cast<detray::scalar>(i0) * step0, 0.f,
+                    static_cast<detray::scalar>(i1) * step1};
+                const auto &v = d.volume_by_pos(rz);
+
+                benchmark::DoNotOptimize(successful);
+                benchmark::DoNotOptimize(unsuccessful);
+                if (v.index() == detray::dindex_invalid) {
+                    ++unsuccessful;
+                } else {
+                    ++successful;
+                }
+                benchmark::ClobberMemory();
+            }
+        }
+    }
+
+#ifdef DETRAY_BENCHMARK_PRINTOUTS
+    std::cout << "Successful   : " << successful << std::endl;
+    std::cout << "Unsuccessful : " << unsuccessful << std::endl;
+#endif  // DETRAY_BENCHMARK_PRINTOUTS
+}
+
+BENCHMARK(BM_FIND_VOLUMES)
+#ifdef DETRAY_BENCHMARK_MULTITHREAD
+    ->ThreadRange(1, benchmark::CPUInfo::Get().num_cpus)
+#endif
+    ->Unit(benchmark::kMillisecond);
```

Comment:
Notice that I registered all of the benchmarks like this now.
  - The multi-threadedness of the benchmark needs to be declared in code. But instead of only running with the maximum number of threads, this setup runs the test with a number of different thread numbers. As explained in:

https://github.com/google/benchmark/blob/main/docs/user_guide.md#multithreaded-benchmarks

  - I stayed with the millisecond printouts, as I agree, this is nicer than the nanosecond numbers that the code would present otherwise.
  - I removed everything else. As all those options can also be supplied to the test from the command line, when executing the benchmark. I didn't see it being worth to hardcode them.